R/register_extendr.R

```diff
@@ -28,23 +28,23 @@
 #' @seealso [rextendr::document()]
 #' @export
 register_extendr <- function(path = ".", quiet = FALSE, force = FALSE, compile = NA) {
-  usethis_quiet <- getOption("usethis.quiet")
-  on.exit(options(usethis.quiet = usethis_quiet))
-  options(usethis.quiet = quiet)
+
+  if (quiet) {
+    withr::local_options(list("cli.default_handler" = function(...) { }))
+  }
```

Comment:
per suggestion of @eitsupi using {withr} to set local options. 

Since usethis ui_ functions are not used in this package and the package is only suggested, I am unsure if the package should be relying on a usethis option to silence output. 

All `usethis.quiet` references have been replaced with cli.

Reply:
Agreed on both counts. Although as a thought, is https://withr.r-lib.org/reference/with_sink.html preferable here because it will silence all output/messages instead of just `cli` ones? Or are we silencing CLI specifically?

Reply:
Great point. I think with_sink() is appropriate for functions like `document()` that call external funcitons. But since, in theory, all messages are being emitted with `cli` they _should be_ functionally equivalent. I'm fine with either.